index.js

```diff
@@ -145,85 +184,4 @@ CachingWriter.prototype.shouldBeIgnored = function (fullPath) {
   return (this._shouldBeIgnoredCache[fullPath] = false);
 };
 
-CachingWriter.prototype.keyForTree = function (fullPath, initialRelativePath, dir) {
-  var relativePath = initialRelativePath || '.';
-  var stats;
-  var statKeys;
-  var type;
-
-  try {
-    this._stats.stats++;
-    stats = fs.statSync(fullPath);
-  } catch (err) {
-    console.warn('Warning: failed to stat ' + fullPath);
-    // fullPath has probably ceased to exist. Leave `stats` undefined and
-    // proceed hashing.
-  }
-
-  var children;
-
-  // has children;
-  if (stats && stats.isDirectory()) {
-    type = 'directory';
-
-    var files;
-
-    try {
-      files = fs.readdirSync(fullPath).sort();
-    } catch (err) {
-      console.warn('Warning: Failed to read directory ' + fullPath);
-      console.warn(err.stack);
-    }
-
-    if (canUseInputFiles(this._inputFiles)) {
-      children = this._inputFiles.map(function(file) {
-        return this.keyForTree(
-          path.join(dir, file),
-          file,
-          dir
-        );
-      }, this);
-      this._stats.files += children.length;
-      children = children.filter(Boolean);
-    } else if (files) {
-      this._stats.files += files.length;
-      children = files.map(function(file) {
-        return this.keyForTree(
-          path.join(fullPath, file),
-          path.join(relativePath, file),
-          dir
-        );
-      }, this).filter(Boolean);
-    }
-
-  } else if (stats && stats.isFile()) {
-    type = 'file';
-
-    if (this.shouldBeIgnored(fullPath)) {
-      return null;
-    }
-  }
-
-  return new Key(type, fullPath, relativePath, stats, children, this.debug());
-};
-
-// Returns a list of matched files
-CachingWriter.prototype.listFiles = function() {
```

Comment:
We may want to bring this back if we can, the use case described in https://github.com/ember-cli/broccoli-caching-writer/pull/40 seems pretty reasonable to me.